tests/thread_float/main.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2017 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief Thread test application
+ *
+ * @author Vincent Dupont <vincent@otakeys.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "thread.h"
+#include "msg.h"
+#include "xtimer.h"
+#include "timex.h"
+
+static char t1_stack[THREAD_STACKSIZE_MAIN];
+static char t2_stack[THREAD_STACKSIZE_MAIN];
+static char t3_stack[THREAD_STACKSIZE_MAIN];
+
+static kernel_pid_t p1, p2, p3;
+
+static xtimer_t timer;
+#define OFFSET (10 * XTIMER_BACKOFF)
+
+static mutex_t lock = MUTEX_INIT;
+
+static void timer_cb(void *arg)
+{
+    (void) arg;
+
+    xtimer_set(&timer, OFFSET);
+    sched_switch(THREAD_PRIORITY_MAIN - 2);
```

Comment:
shouldn't "thread_yield()``` do the trick here?

Reply:
I tested with thread_yield() and it seems to work ok!
```
2018-06-01 15:58:12,754 - INFO # THREADS CREATED
2018-06-01 15:58:12,755 - INFO # 
2018-06-01 15:58:12,756 - INFO # THTTRHHERREEAAADDD   55 5 s ststatraatrr
2018-06-01 15:58:12,757 - INFO # tt
2018-06-01 15:58:12,758 - INFO # 
2018-06-01 15:58:12,898 - INFO # T(5): 141.521576
2018-06-01 15:58:12,899 - INFO # T(3): 141.466812
2018-06-01 15:58:13,042 - INFO # T(5): 141.559769
2018-06-01 15:58:13,049 - INFO # T(3): 141.490616
2018-06-01 15:58:13,192 - INFO # T(5): 141.605148
2018-06-01 15:58:13,198 - INFO # T(3): 141.521576
2018-06-01 15:58:13,343 - INFO # T(5): 141.657516
2018-06-01 15:58:13,344 - INFO # T(3): 141.559769
2018-06-01 15:58:13,492 - INFO # T(5): 141.717255
2018-06-01 15:58:13,494 - INFO # T(3): 141.605148
```